addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/apachev5/HttpSenderApache.java

```diff
@@ -0,0 +1,569 @@
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.addon.network.ClientCertificatesOptions;
+import org.zaproxy.addon.network.ConnectionOptions;
+import org.zaproxy.addon.network.internal.client.BaseHttpSender;
+import org.zaproxy.addon.network.internal.client.LegacyUtils;
+import org.zaproxy.addon.network.internal.client.ResponseBodyConsumer;
+import org.zaproxy.addon.network.internal.client.SocksProxy;
+import org.zaproxy.zap.ZapGetMethod;
+import org.zaproxy.zap.network.HttpRequestConfig;
+import org.zaproxy.zap.users.User;
+
+/** A {@link BaseHttpSender} using Apache HttpComponents Client. */
+public class HttpSenderApache
+        extends BaseHttpSender<HttpSenderContextApache, ZapHttpClientContext, HttpEntity> {
+
+    private static final Logger LOGGER = LogManager.getLogger(HttpSenderApache.class);
+
+    private final Supplier<CookieStore> globalCookieStoreProvider;
+    private final ConnectionOptions options;
+
+    private final Lookup<AuthSchemeFactory> authSchemeRegistry;
+
+    private final ProxyRoutePlanner routePlanner;
+    private final ProxyCredentialsProvider credentialsProvider;
+    private final RequestConfig defaultRequestConfig;
+    private final ManagedHttpClientConnectionFactory managedHttpClientConnectionFactory;
+    private final OutgoingContentStrategy outgoingContentStrategy;
+    private final LayeredConnectionSocketFactory sslSocketFactory;
+
+    private final PoolingHttpClientConnectionManager connectionManager;
+    private final HttpProcessor proxyHttpProcessor;
+    private final ZapRequestAddCookies zapRequestAddCookies;
+    private final HttpProcessor mainHttpProcessor;
+    private final RequestRetryStrategy requestRetryStrategy;
+    private final CloseableHttpClient clientImpl;
+    private final HttpConnector httpConnector;
+
+    public HttpSenderApache(
+            Supplier<CookieStore> globalCookieStoreProvider,
+            ConnectionOptions options,
+            ClientCertificatesOptions clientCertificatesOptions) {
+        this.globalCookieStoreProvider = Objects.requireNonNull(globalCookieStoreProvider);
+        this.options = Objects.requireNonNull(options);
+
+        authSchemeRegistry =
+                RegistryBuilder.<AuthSchemeFactory>create()
+                        .register(StandardAuthScheme.BASIC, BasicSchemeFactory.INSTANCE)
+                        .register(StandardAuthScheme.DIGEST, DigestSchemeFactory.INSTANCE)
+                        .register(StandardAuthScheme.NTLM, NTLMSchemeFactory.INSTANCE)
+                        .register(StandardAuthScheme.SPNEGO, SPNegoSchemeFactory.DEFAULT)
+                        .register(StandardAuthScheme.KERBEROS, KerberosSchemeFactory.DEFAULT)
```

Comment:
Does this mean we have more modern Microsoft'ish auth handling?

Reply:
Maybe, I didn't try it but it does support more authentication schemes.